DNN-OpenCV-Classification-with-Java/README.md

```diff
@@ -0,0 +1,197 @@
+```
+
+### OpenCV
+
+In this blog post we are using OpenCV 4.3.0 unavailable via `pip` and OpenCV for Java. That is why, we first need to build the
+OpenCV library. To do so:
+
+1. Check the list of the below libraries. Install the missed dependencies:
+
+```bash
+sudo apt-get update
+sudo apt-get install build-essential cmake unzip pkg-config
+sudo apt-get install libjpeg-dev libpng-dev libtiff-dev
+sudo apt-get install libavcodec-dev libavformat-dev libswscale-dev libv4l-dev
+sudo apt-get install libxvidcore-dev libx264-dev
+sudo apt-get install libgtk-3-dev
+sudo apt-get install libatlas-base-dev gfortran
+sudo apt-get install python3-dev
+```
+
+For OpenCV Java installation we used default Java Runtime Environment and Java Development Kit:
+
+```bash
+sudo apt-get install default-jre
+sudo apt-get install default-jdk
+sudo apt-get install ant
+```
+
+2. Download the latest OpenCV version from the official repository:
+
+```bash
+cd ~
+wget -O opencv.zip https://github.com/opencv/opencv/archive/4.3.0.zip
+```
+
+3. Unzip the downloaded archives:
+
+```bash
+unzip opencv.zip
+```
+
+4. Rename the directories to match CMake paths:
+
+```bash
+mv opencv-4.3.0 opencv
+```
+
+5. Compile OpenCV. Create and enter a build directory:
+
+```bash
+cd ~/opencv
+mkdir build
+```
+
+6. Run CMake to configure the OpenCV build. Don't forget to set the right pass to the ``PYTHON_EXECUTABLE``:
+
+```bash
+cmake -D CMAKE_BUILD_TYPE=RELEASE \
+ -D CMAKE_INSTALL_PREFIX=/usr/local \
+ -D INSTALL_PYTHON_EXAMPLES=OFF \
+ -D INSTALL_C_EXAMPLES=OFF \
+ -D OPENCV_ENABLE_NONFREE=ON \
+ -D OPENCV_EXTRA_MODULES_PATH=~/opencv_contrib/modules \
+ -D PYTHON_EXECUTABLE=~/env/bin/python3 \
+ -D BUILD_SHARED_LIBRARY=OFF \
+ -D BUILD_TESTS=OFF \
+ -D BUILD_PERF_TESTS=OFF \
+ -D BUILD_EXAMPLES=ON ..
+```
+
+If you want to configure the build with some specific Java version, please, add the following fields, verifying the paths:
+
+```bash
+ -D JAVA_AWT_INCLUDE_PATH=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include \
+ -D JAVA_AWT_LIBRARY=/usr/lib/jvm/java-1.x.x-openjdk-amd64/lib/libawt.so \
+ -D JAVA_INCLUDE_PATH=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include \
+ -D JAVA_INCLUDE_PATH2=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include/linux \
+ -D JAVA_JVM_LIBRARY=/usr/lib/jvm/java-1.x.x-openjdk-amd64/include/jni.h \
+```
+
+7. Check the output and make sure that everything is set correctly. After that we're ready to build it with:
+
+```bash
+make -j8
+```
+
+Make sure, you didn't get any errors. In case of successful completion you will find the following files in the ``build`` directory:
+* ``bin/opencv-430.jar``
+* ``lib/libopencv_java430.so``
+* ``lib/python3/cv2.cpython-37m-x86_64-linux-gnu.so``
+
+Then run the following command:
+
+```bash
+sudo ldconfig
+```
+
+which creates the necessary links and cache to our freshly built shared library.
+
+Rename the created Python3 bindings for OpenCV to `cv2.so`:
+
+```bash
+mv lib/python3/cv2.cpython-37m-x86_64-linux-gnu.so cv2.so
```

Comment:
I propose not to rename so file as it contains python version. You just need to put *.so file to proper location or add it to PYTHONPATH.